django_cte/meta.py

```diff
@@ -75,21 +75,8 @@ class CTEColumnRef(Expression):
     def __init__(self, name, output_field):
         self.name = name
         self.output_field = output_field
-        self._alias = None
-
-    def resolve_expression(self, query=None, allow_joins=True, reuse=None,
-                           summarize=False, for_save=False):
-        ref = self.copy()
-        ref._alias = query.get_initial_alias()
-        return ref
-
-    def relabeled_clone(self, relabels):
-        if self._alias is not None and self._alias in relabels:
-            clone = self.copy()
-            clone._alias = relabels[self._alias]
-            return clone
-        return self
```

Comment:
I'm reluctant to merge this without digging a bit deeper to understand how this method can be safely removed. If it turns out to be unnecessary then this looks like an elegant fix. I see that all tests have passed, which is a good sign. However, I seem to remember some edge cases where `relabeled_clone` was necessary (although I think I would have added tests for them). Can you explain in a bit more detail why this is no longer needed? Out of curiosity, is `relabeled_clone` no longer a thing in Django 3?

Even if this works perfectly in Django 3, I'm concerned that it may break backward compatibility with earlier versions (although once again, should have been caught by tests since we test against older versions of Django). Can you explain why that is not a concern?

Reply:
Unfortunately my answer is the one you're hoping for. I removed it since it was changing the `_alias` property that was used when generating the sql is no longer necessary. Since there's no need for a mutable property to keep track of the alias, there's no reason to change it elsewhere. I'm not familiar with the internals of the Django ORM, but I did test this against my application before creating the PR.

Do you know of a case in which relabeled_clone is called / used? If you can provide me with the general idea of it, I can make sure it's tested.

Reply:
I will see if I can come up with a test that breaks it. If I can't I'll consider it good and merge.

Reply:
Thank you, @millerdev !

Reply:
I've been unable to write a failing test. In more detail (mostly for personal reference in case this ever becomes an issue later), I'm unable to find a scenario where `compiler.query.get_initial_alias()` returns the wrong result for a `CTEColumnRef`. It's hard to know all the possible values of `compiler.query`, but it looks like should always be a `CTEQuery`, given that `CTEColumnRef` is only created by `With.queryset()`:

https://github.com/dimagi/django-cte/blob/e999216ce2e0dd784060a8285bfe9470a867f683/django_cte/cte.py#L105-L111

Reply:
I ran some more tests and found a [failure](https://github.com/dimagi/commcare-hq/blob/0d0f6c238d64d52ceab07cf1c3ab82742a4e429e/corehq/apps/locations/tests/test_location_utils.py#L42)

```
======================================================================
FAIL: corehq.apps.locations.tests.test_location_utils:TestGetLocationsAndChildren.test_get_locations_and_children
----------------------------------------------------------------------
Traceback (most recent call last):
  File ".../corehq/apps/locations/tests/test_location_utils.py", line 49, in test_get_locations_and_children
    ['Middlesex', 'Cambridge', 'Somerville', 'Suffolk', 'Boston']
AssertionError: Element counts were not equal:
First has 2, Second has 1:  'Somerville'
```

I believe it is a result of this change.

Good SQL snippet (django-cte==1.1.4)
```sql
EXISTS (
    SELECT U0."id", U0."_cte_ordering" AS "_cte_ordering"
    FROM "xdups" U0
    WHERE (
		U0."id" = (cte."id")
		AND U0."_cte_ordering" = (cte."_cte_ordering")
	)
) AS "_exclude_dups"
```

Bad SQL snippet (this PR)
```sql
EXISTS (
    SELECT U0."id", U0."_cte_ordering" AS "_cte_ordering"
    FROM "xdups" U0
    WHERE (
    	U0."id" = (cte."id")
    	AND U0."_cte_ordering" = (U0."_cte_ordering")   -- NOTE wrong RHS alias
	)
) AS "_exclude_dups"
```

Dependency versions:
- Django 1.11.27
- Python 3.7.2

Reply:
This doesn't appear to be related to removing `relabed_clone`, but rather another symptom of the Django 3 changes that caused the original implementation to fail.

Reply:
@millerdev : Does his response alleviate your concern for this particular pull request? If so, perhaps the right course of action would be to merge this, remove the "close # 10" from the description of this pull request, and create another issue for that?

Reply:
Actually the latest changes _should_ fix the issue @millerdev brought up. I was able to add a test for that specific case, then fix the problem.

Reply:
Awesome! Thanks for clarifying that.

Reply:
I plan to run external tests against this again and make a decision based on that outcome. I'll try to do that soon.